src/graphics/reproject-texture.js

```diff
@@ -42,15 +43,48 @@ function getCoding(texture) {
  * the source is convolved by a phong-weighted kernel raised to the specified power. Otherwise
  * the function performs a standard resample.
  * @param {number} [numSamples] - optional number of samples (default is 1024).
+ * @param {number} [sourceProjection] - optional source projection. Can be:
+ *
+ * * {@link pc.TEXTUREPROJECTION_CUBE}
+ * * {@link pc.TEXTUREPROJECTION_EQUIRECT}
+ * * {@link pc.TEXTUREPROJECTION_OCTAHEDRAL}
+ *
+ * @param {number} [targetProjection] - optional target projection. Can be:
+ *
+ * * {@link pc.TEXTUREPROJECTION_CUBE}
+ * * {@link pc.TEXTUREPROJECTION_EQUIRECT}
+ * * {@link pc.TEXTUREPROJECTION_OCTAHEDRAL}
  */
-function reprojectTexture(device, source, target, specularPower, numSamples = 1024) {
-    var processFunc = (specularPower !== undefined) ? 'prefilter' : 'reproject';
-    var decodeFunc = "decode" + getCoding(source);
-    var encodeFunc = "encode" + getCoding(target);
-    var sourceFunc = source.cubemap ? "sampleCubemap" : "sampleEquirect";
-    var targetFunc = target.cubemap ? "getDirectionCubemap" : "getDirectionEquirect";
+function reprojectTexture(device, source, target, specularPower, numSamples = 1024, sourceProjection, targetProjection) {
```

Comment:
Should the last two params have defaults?

Reply:
Oh, and it looks like `specularPower` should have a default parameter of 1.

Reply:
done both. I've updated docs as well - default values are two, depending on what the texture type is. I guess that's ok.